Doc/c-api/object.rst

```diff
@@ -85,6 +85,28 @@ Object Protocol
    instead of the :func:`repr`.
 
 
+.. c:function:: void PyObject_Dump(PyObject* op)
+
+   Dump an object *op* to ``stderr``. This should only be used for debugging.
+
+   This function can be called without an :term:`attached thread state`, but it's not
+   recommended to do so.
+
+   Implement a heuristic to detect if the object memory has been freed.
+
+   Example of output:
```

Comment:
```suggestion
   The output is intended to try dumping objects even after memory corruption:

   - Information is written starting with fields that are the least likely to crash when accessed
   - Less information is printed if Python detects that the object has been freed
   - This function can be called without an :term:`attached thread state`, but this may cause race conditions, deadlocks or crashes
   - An object that does not belong to the current interpreter may be dumped, but this may also cause crashes or unintended behavior
   
   The output format may change at any time.

   Example of output:
```